test/protocol/OfferHandlerTest.js

```diff
@@ -1264,6 +1276,198 @@ describe("IBosonOfferHandler", function () {
+      });
+
+      context("💔 Revert Reasons", async function () {
+        it("The offers region of protocol is paused", async function () {
+          // Pause the offers region of the protocol
+          await pauseHandler.connect(pauser).pause([PausableRegion.Offers]);
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(operator).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.REGION_PAUSED
+          );
+        });
+
+        it("The exchanges region of protocol is paused", async function () {
+          // Pause the exchanges region of the protocol
+          await pauseHandler.connect(pauser).pause([PausableRegion.Exchanges]);
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(operator).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.REGION_PAUSED
+          );
+        });
+
+        it("Offer does not exist", async function () {
+          // Set invalid id
+          id = "444";
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(operator).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.NO_SUCH_OFFER
+          );
+
+          // Set invalid id
+          id = "0";
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(operator).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.NO_SUCH_OFFER
+          );
+        });
+
+        it("Offer already voided", async function () {
+          // Void the offer first
+          await offerHandler.connect(operator).voidOffer(id);
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(operator).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.OFFER_HAS_BEEN_VOIDED
+          );
+        });
+
+        it("Caller is not seller", async function () {
+          // caller is not the operator of any seller
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(rando).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.NOT_OPERATOR
+          );
+
+          // caller is an operator of another seller
+          // Create a valid seller, then set fields in tests directly
+          seller = mockSeller(rando.address, rando.address, rando.address, rando.address);
+
+          // AuthToken
+          emptyAuthToken = mockAuthToken();
+          expect(emptyAuthToken.isValid()).is.true;
+          await accountHandler.connect(rando).createSeller(seller, emptyAuthToken, voucherInitValues);
+
+          // Attempt to reserve a range, expecting revert
+          await expect(offerHandler.connect(rando).reserveRange(id, length)).to.revertedWith(
+            RevertReasons.NOT_OPERATOR
+          );
+        });
+
+        it("Range length is zero", async function () {
```

Comment:
Could you please also add the test to check if range is lower than 1 << 128

Reply:
<img width="644" alt="Screen Shot 2022-12-14 at 15 02 05" src="https://user-images.githubusercontent.com/50335980/207672059-fe84602d-f964-4b7f-aebe-21389b53baa9.png">

that's odd, coverage reported as a uncoverage branch

Reply:
Oh, yes.
I need to adjust `offer.quantityAvailable` to be greater than (1<<128) before I test for revert reason, since now it reverts 1 require too soon.
I'll do it tomorrow.

Reply:
done